azurerm_virtual_machine_data_disk_attachment crashes if attaching are shared disk via a module

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Terraform (and AzureRM Provider) Version
Terraform v0.15.5
azurerm v2.96.0

<!--- Please run `terraform -v` to show the Terraform core version and provider version(s). If you are not running the latest version of Terraform or the provider, please upgrade because your issue may have already been fixed. [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions). --->

### Affected Resource(s)

<!--- Please list the affected resources and data sources. --->

* `azurerm_virtual_machine_data_disk_attachment`

### Terraform Configuration Files

<!--- Information about code formatting: https://help.github.com/articles/basic-writing-and-formatting-syntax/#quoting-code --->

```hcl
#from within the module:
resource "azurerm_virtual_machine_data_disk_attachment" "attach_data_disk" {
  for_each           = {for disk in var.disks:disk.name => disk}
  managed_disk_id    = azurerm_managed_disk.data[each.key].id 
  virtual_machine_id = azurerm_windows_virtual_machine.virtual_machine.id
  lun                = each.value.lun
  caching            = "${each.value.type == "UltraSSD_LRS" ? "None" : each.value.max-shares > "0" ? "None" :"ReadWrite" }"
}

#output is done via:

output "managed_disk_lun" {

value = {for k, v in azurerm_virtual_machine_data_disk_attachment.attach_data_disk : k=> tomap({id=v.id, lun=v.lun}) }
}

#from within the testvm.tf:

resource "azurerm_virtual_machine_data_disk_attachment" "attach_shared_disks_cl2" {
  managed_disk_id    = module.cl1.managed_disk_lun.E.id
  virtual_machine_id = module.cl2.vm_id
  lun                = module.cl1.managed_disk_lun.E.lun
  caching            = "None"
}
```

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.

To obtain the debug output, see the [Terraform documentation on debugging](https://www.terraform.io/docs/internals/debugging.html).
--->
```hcl
### Panic Output
azurerm_virtual_machine_data_disk_attachment.attach_shared_disks_cl2: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with azurerm_virtual_machine_data_disk_attachment.attach_shared_disks_cl2,
│   on vm_module_test.tf line 62, in resource "azurerm_virtual_machine_data_disk_attachment" "attach_shared_disks_cl2":
│   62: resource "azurerm_virtual_machine_data_disk_attachment" "attach_shared_disks_cl2" {
│ 
│ The plugin encountered an error, and failed to respond to the
│ plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵
Stack trace from the terraform-provider-azurerm_v2.96.0_x5 plugin:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x47e9540]
goroutine 118 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/compute.retrieveDataDiskAttachmentManagedDisk(0x664a640, {0x5686660, 0xc0014d2a80}, {0xc0000742c0, 0xa9})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/compute/virtual_machine_data_disk_attachment_resource.go:304 +0x100
github.com/hashicorp/terraform-provider-azurerm/internal/services/compute.resourceVirtualMachineDataDiskAttachmentCreateUpdate(0xc0001bd900, {0x5686660, 0xc0014d2a80})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/compute/virtual_machine_data_disk_attachment_resource.go:117 +0x72b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0019bd6b0, {0x669d2d8, 0xc001cc7b00}, 0x2, {0x5686660, 0xc0014d2a80})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:329 +0x178
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00046c8c0, {0x669d2d8, 0xc001cc7b00}, 0xc0006b5ba0, 0xc0001bd780, {0x5686660, 0xc0014d2a80})
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:467 +0x871
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0002b6060, {0x669d2d8, 0xc001cc7b00}, 0xc00051aa00)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000[298](https://git.dpd.de/azure-infrastructure/automation-testing-windows/win-infrastructure/-/jobs/434586#L298)500, {0x669d380, 0xc0019bc030}, 0xc00194a150)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:603 +0x30e
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x5bd49a0, 0xc000298500}, {0x669d380, 0xc0019bc030}, 0xc001d1d680, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000236700, {0x66ecd18, 0xc0003ce480}, 0xc0019faa20, 0xc000c45a40, 0xacff1e0, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0xc6f
google.golang.org/grpc.(*Server).handleStream(0xc000236700, {0x66ecd18, 0xc0003ce480}, 0xc0019faa20, 0x0)
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x294
Error: The terraform-provider-azurerm_v2.96.0_x5 plugin crashed!
This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
Cleaning up project directory and file based variables
00:02
ERROR: Job failed: exit code 1

```
<!--- If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. --->

### Expected Behaviour
The exisisting shared disk of cl1 with maxhshares=2 value should be attached to the exisiting vm cl2.
<!--- What should have happened? --->

### Actual Behaviour
Terraform Plan works fine, Terraform Apply crashes.
<!--- What actually happened? --->

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

1. `terraform apply`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? For example: Running in a Azure China/Germany/Government? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Such as vendor documentation?
--->

* #0000

--- a/internal/services/compute/virtual_machine_data_disk_attachment_resource.go
+++ b/internal/services/compute/virtual_machine_data_disk_attachment_resource.go
@@ -301,7 +301,7 @@ func retrieveDataDiskAttachmentManagedDisk(d *pluginsdk.ResourceData, meta inter
 
 	parsedId, err := parse.ManagedDiskID(id)
 	if err != nil {
-		return nil, fmt.Errorf("parsing Managed Disk ID %q: %+v", parsedId.String(), err)
+		return nil, fmt.Errorf("parsing Managed Disk ID %q: %+v", id, err)
 	}
 
 	resp, err := client.Get(ctx, parsedId.ResourceGroup, parsedId.DiskName)

